internal/cli/atlas/serverlessclusters/watch.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverlessclusters
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	name  string
+	store store.ServerlessInstanceDescriber
+}
+
+func (opts *WatchOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.ServerlessInstance(opts.ConfigProjectID(), opts.name)
+	if err != nil {
+		return false, err
+	}
+	return result.StateName == "IDLE", nil
+}
+
+func (opts *WatchOpts) Run() error {
+	if err := opts.Watch(opts.watcher); err != nil {
+		return err
+	}
+
+	return opts.Print(nil)
+}
+
+// mongocli atlas serverlessCluster(s) watch <clusterName> [--projectId projectId].
+func WatchBuilder() *cobra.Command {
+	opts := &WatchOpts{}
+	cmd := &cobra.Command{
+		Use:   "watch <clusterName>",
+		Short: "Watch for a serverless cluster to be available.",
+		Args:  require.ExactArgs(1),
+		Annotations: map[string]string{
+			"args":            "clusterName",
+			"clusterNameDesc": "Name of the cluster to watch.",
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(
+				opts.ValidateProjectID,
```

Comment:
[q] Is `ValidateProjectID` correct? (all other references in the file are to `projectId`

Reply:
yes, it is. We validate the project id inside the `opts.PreRunE` also in the other command.
`ValidateProjectID` checks that one the user has provided the `--projectId` flag or that the used profile has the `projectId`

```golang
// ValidateProjectID validates projectID.
func (opts *GlobalOpts) ValidateProjectID() error {
	if opts.ConfigProjectID() == "" {
		return errMissingProjectID
	}
	return validate.ObjectID(opts.ConfigProjectID())
}

// ConfigProjectID returns the project id.
// If the id is empty, it caches it after querying config.
func (opts *GlobalOpts) ConfigProjectID() string {
	if opts.ProjectID != "" {
		return opts.ProjectID
	}
	opts.ProjectID = config.ProjectID()
	return opts.ProjectID
}
```

Reply:
I _think_ it is correct. It is defined in https://github.com/mongodb/mongocli/blob/fcf9f4ee31467445cdbe80cbdca1392444ea2bde/internal/cli/global_opts.go#L66

And all the methods there end with `ProjectID`. Maybe it is just a styling issue.